pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py

```diff
@@ -25,13 +25,20 @@
 import torch
 import numpy as np
 
-from pytorch_pretrained_bert.modeling import BertConfig, BertForPreTraining, load_tf_weights_in_bert
+from pytorch_pretrained_bert.modeling import BertConfig, BertForQuestionAnswering, BertForPreTraining, \
+    load_tf_weights_in_bert
 
-def convert_tf_checkpoint_to_pytorch(tf_checkpoint_path, bert_config_file, pytorch_dump_path):
+
+def convert_tf_checkpoint_to_pytorch(tf_checkpoint_path, bert_config_file, pytorch_dump_path, bert_model):
     # Initialise PyTorch model
     config = BertConfig.from_json_file(bert_config_file)
     print("Building PyTorch model from configuration: {}".format(str(config)))
-    model = BertForPreTraining(config)
+
+    # Dynamically choose the model type based on input type
+    if bert_model == "qa":
+        model = BertForQuestionAnswering(config)
```

Comment:
Maybe we could use a more general way to give the class name as a string, like this:
```python
import importlib
ModelClass = getattr(importlib.import_module("pytorch_pretrained_bert.modeling"),
                     bert_model)
model = ModelClass(config)
```
See this: https://stackoverflow.com/questions/4821104/python-dynamic-instantiation-from-string-name-of-a-class-in-dynamically-imported

Reply:
My concern with that approach, which indeed is more generic, was the case of having different constructors, e.g. "BertForTokenClassification(config, 2)" (which is the only model, atm). I guess we can do dynamic argument passing as well.

Something like this: [How can you set class attributes from variable arguments (kwargs) in python](https://stackoverflow.com/questions/8187082/how-can-you-set-class-attributes-from-variable-arguments-kwargs-in-python)

Maybe it's better to move the bert_model (instance class name) and constructor arguments into a new config file (e.g. converter_config.json)?

`{"bert_model": "BertForTokenClassification",
"num_labels": 2,
"bert_config_file": "/some/path/here"}`

Several questions:
1) I'm not sure if it's not an overkill, since just one class at the moment has an extra argument in the constructor?
2) Is it a good idea to move bert_config_file into the "config" file?
3) Should we move then all other args to the config?
4) Should we allow "default" execution without converter_config, that instantiates "BertForPreTraining"?